googlemaps/client.py

```diff
@@ -175,10 +187,22 @@ def _get(self, url, params, first_request_time=None, retry_counter=0,
             return self._get(url, params, first_request_time, retry_counter + 1,
                              base_url, accepts_clientid, extract_body)
 
+        # If `queries_per_second` was configured and at least that many
+        # queries have been made, check if the time of the earliest query
+        # is under a second ago - if so, sleep for the difference between
+        # it, and a second.
+        if self.sent_times and len(self.sent_times) == self.sent_times.maxlen:
+            elapsed_since_earliest = time.time() - self.sent_times[0]
+            if elapsed_since_earliest < 1:
+                time.sleep(1 - elapsed_since_earliest)
+
```

Comment:
Hm, good point :-)

It might be possible though that a client could be shared across Python threads, in which case round trip times wouldn't be the limiting factor.